src/test/java/org/opensearch/knn/integ/QueryParseIT.java

```diff
@@ -0,0 +1,148 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.knn.integ;
+
+import lombok.SneakyThrows;
+import org.opensearch.client.Request;
+import org.opensearch.client.ResponseException;
+import org.opensearch.common.xcontent.XContentFactory;
+import org.opensearch.core.xcontent.XContentBuilder;
+import org.opensearch.knn.KNNRestTestCase;
+
+import java.io.IOException;
+import java.util.Locale;
+
+import static org.opensearch.knn.index.query.parser.RescoreParser.RESCORE_OVERSAMPLE_PARAMETER;
+import static org.opensearch.knn.index.query.parser.RescoreParser.RESCORE_PARAMETER;
+
+public class QueryParseIT extends KNNRestTestCase {
```

Comment:
nit: https://github.com/opensearch-project/k-NN/blob/main/src/test/java/org/opensearch/knn/index/query/InvalidSearchQueryIT.java 

Seems like a duplicate in a way, maybe added a todo on one of them to merge for later

Reply:
Yes, that probably makes sense. Let me make a note on this and come back with it in another PR, with some other test cleanup.